block/manager.go

```
@@ -1492,3 +1500,127 @@
 
 	return nil
 }
+
+// Handshake performs the ABCI handshake with the application.
+func (m *Manager) Handshake(ctx context.Context, proxyApp proxy.AppConns) error {
+	// Handshake is done via ABCI Info on the query conn.
+	res, err := proxyApp.Query().Info(ctx, proxy.RequestInfo)
+	if err != nil {
+		return fmt.Errorf("error calling Info: %v", err)
```

golangci-lint (1.62.2)

non-wrapping format verb for fmt.Errorf. Use `%w` to format errors

(errorlint)